src/native/eventpipe/ep-thread.c

```diff
@@ -247,7 +247,6 @@ ep_thread_get_session_state (
 
 	ep_thread_requires_lock_held (thread);
 
-	EP_ASSERT (thread->session_state [ep_session_get_index (session)] != NULL);
```

Comment:
Did you hit this assert? Looking at the callers of ep_thread_get_session_state, you should end up with more asserts in the case this returns NULL

Reply:
The code in ep_session_remove_dangling_session_states iterates over all threads and under normal circumstances it will return NULL, it will only return non-NULL if a session state is leaked. So we will hit this assert in the code I added

Reply:
Great! I didn't find your call since I didn't search your branch for callers of ep_thread_get_session_state and for the other once it is critical that this assert holds. Maybe we could move out the assert to the callers where it should still be true?